src/hooks/useMultiSelect.stories.mdx

```diff
@@ -0,0 +1,255 @@
+
+```jsx
+const multiSelectItems = [
+  {
+    id: "pets",
+    name: "Pets",
+    items: [
+      { id: "cat", name: "Cat" },
+      {
+        id: "dog",
+        name: "Dog",
+        children: [
+          { id: "corgy", name: "Corgy" },
+          { id: "german-sheperd", name: "German Sheperd" },
+          { id: "afghan-hound", name: "Afghan Hound" },
+        ],
+      },
+      { id: "rat", name: "Rat" },
+    ],
+  },
+  {
+    id: "tools",
+    name: "Tools",
+    items: [
+      { id: "hammer", name: "Hammer" },
+      {
+        id: "screwdriver",
+        name: "Screwdriver",
+        children: [
+          { id: "slottet", name: "Slotted Screwdriver" },
+          { id: "phillips", name: "Phillips Screwdriver" },
+          { id: "allen", name: "Allen Screwdriver" },
+        ],
+      },
+      { id: "whisk", name: "Whisk" },
+    ],
+  },
+];
+```
+
+The implemntation of the `useMultiSelect` hook together with the `MultiSelectGroup` component.
+
+```jsx
+import React from "react";
+import {
+  MultiSelect,
+  MultiSelectGroup,
+  useMultiSelect,
+} from "@nypl/design-system-react-components";
+export default function MultiSelectGroupWithHookExample() {
+  const { onChange, onMixedStateChange, onClear, selectedItems } =
+    useMultiSelect();
+  return (
+    <MultiSelectGroup id="multiselect-group" labelText="MultiSelect Group">
+      {multiSelectItems &&
+        multiSelectItems.map((multiSelect) => (
+          <MultiSelect
+            key={multiSelect.id}
+            id={multiSelect.id}
+            label={multiSelect.name}
+            variant="dialog"
+            items={multiSelect.items}
+            selectedItems={selectedItems}
+            onChange={(e) => {
+              onChange(e.target.id, multiSelect.id);
+            }}
+            onMixedStateChange={(e) =>
+              onMixedStateChange(e.target.id, multiSelect.id, multiSelect.items)
+            }
+            onClear={() => {
+              onClear(multiSelect.id);
+            }}
+            onApply={() => {}}
+          />
+        ))}
+    </MultiSelectGroup>
+  );
+  );
+}
+```
+
+## setSelectedItems
+
+The data structure to pass the setSelectedItems function in order to set a inital state:
+
+```jsx
+// The selectedItems structur for one MultiSelect
+const initialSelectedItems = {
+  colors: {
+    items: ["red"],
+  },
+};
+
+// The selectedItems structur for setting state of multiple MultiSelects
+const initialSelectedItems = {
+  colors: {
+    items: ["red"],
+  },
+  pets: {
+    items: ["corgy", "german-sheperd"],
+  },
+};
+```
+
+```jsx
+import React from "react";
+import {
+  MultiSelect,
+  MultiSelectGroup,
+  useMultiSelect,
+} from "@nypl/design-system-react-components";
+export default function MultiSelectGroupWithHookExample() {
+  const { onChange, onMixedStateChange, onClear, selectedItems, setSelectedItems } =
+    useMultiSelect();
+
+  React.useEffect(() => {
+    setSelectedItems(initialSelectedItems)
+  }, [])
```

Comment:
The example in `useFilterBar.ts` does not use the `useEffect`. Make it consistent for both examples.

Reply:
Added useEffect since in Scout we have to use it it this way in order to work.